### ISSUE TITLE
Strange recursive stack overflow when emitting messages from the disconnect handler.

### ISSUE THREAD

**Issue by fake-name** [miguelgrinberg/Flask-SocketIO#230]
I'm not in any way confident I'll find a way to reproduce, or properly debug this issue (or even if it is an issue), but it was interesting, and it did entirely take down the web-app I'm working on, so I think it bears at least some discussion.

Basically, I have handlers attached to the `connect` and `disconnect` events:

```
@socketio.on('disconnect', namespace='/vna_interface')
def event_disconnect():
	log.info("event_disconnect() -> %s", request.sid)
	active = app.config['LOGIN_ARBITER'].remove_item(request.sid)
	emit('active', list(active), broadcast=True, namespace='/vna_interface')
```

The LOGIN_ARBITER is a unique-user tracking mechanism, it's not too important here. The relevant bit is that it returns a 2-tuple `(int, int)`.

Anyways, I had my webapp crash with the following traceback:

```
Fatal Python error: Cannot recover from stack overflow.

Current thread 0x00007f632751a740 (most recent call first):
  File "/www/flask/lib/python3.4/site-packages/werkzeug/routing.py", line 1241 in bind
  File "/www/flask/lib/python3.4/site-packages/werkzeug/routing.py", line 1327 in bind_to_environ
  File "/www/flask/lib/python3.4/site-packages/flask/app.py", line 1602 in create_url_adapter
  File "/www/flask/lib/python3.4/site-packages/flask/ctx.py", line 224 in __init__
  File "/www/flask/lib/python3.4/site-packages/flask/app.py", line 1773 in request_context
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 175 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 183 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 183 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 183 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 183 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 183 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 183 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  ...

```

This is a bit interesting, because it doesn't appear to be a normal python traceback. 

line 183 is `emit('active', list(active), broadcast=True, namespace='/vna_interface')`. That's the total extent of my code involved here, I think. 

I'm not entirely sure what happened, and I have *not* been able to persuade my webapp to repeat it, so it may just be a non-issue. It was, however, sufficiently odd I figure asking if anyone has seen anything like this could be helpful.

---

For the time being, I've simply modified the disconnect handler to not emit messages if it doesn't think there are any users connected anymore, and either it fixes the issue, or it just hasn't been willing to replicate the problem.

**Comment by miguelgrinberg** [comment]
Yes, it's an odd one. Are you using the Javascript client or some other client?

**Comment by fake-name** [comment]
JS Client. I'm not doing anything too odd, either, I think.

**Comment by miguelgrinberg** [comment]
I think I have found what the problem is. You are sending a broadcast, and the socket that was disconnected is apparently included and accessed. When I try to emit to that socket I find that it is closed, so I think that the socket has died, so I call the disconnect handler again. And that blows the stack. It's a bug, but you can easily workaround it by telling your emit to not send to `request.pid`.

**Comment by fake-name** [comment]
Sweet. Thanks for the ridiculously fast diagnosis.

**Comment by fake-name** [comment]
Just ran into this again, with some more logging. I can confirm it's a recursion thing in `event_disconnect()`:

```
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Main.Web.MainThread - INFO - event_disconnect() -> 2578cde6e1e249b2b61b933d93107b7f
Fatal Python error: Cannot recover from stack overflow.

Current thread 0x00007f1916c31740 (most recent call first):
  File "/www/flask/lib/python3.4/site-packages/werkzeug/routing.py", line 1241 in bind
  File "/www/flask/lib/python3.4/site-packages/werkzeug/routing.py", line 1327 in bind_to_environ
  File "/www/flask/lib/python3.4/site-packages/flask/app.py", line 1602 in create_url_adapter
  File "/www/flask/lib/python3.4/site-packages/flask/ctx.py", line 224 in __init__
  File "/www/flask/lib/python3.4/site-packages/flask/app.py", line 1773 in request_context
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 175 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit  
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 187 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 187 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 187 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 187 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 96 in close
  File "/www/flask/lib/python3.4/site-packages/engineio/socket.py", line 58 in send
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 158 in send
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 333 in _send_packet
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 322 in _emit_internal
  File "/www/flask/lib/python3.4/site-packages/socketio/base_manager.py", line 103 in emit
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 181 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 282 in emit
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 508 in emit
  File "/www/VNA_Applications/WebVna/app/views.py", line 187 in event_disconnect
  File "/www/flask/lib/python3.4/site-packages/flask_socketio/__init__.py", line 187 in _handler
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 397 in _trigger_event
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 358 in _handle_disconnect
  File "/www/flask/lib/python3.4/site-packages/socketio/server.py", line 444 in _handle_eio_disconnect
  File "/www/flask/lib/python3.4/site-packages/engineio/server.py", line 327 in _trigger_event
```

**Comment by miguelgrinberg** [comment]
@fake-name sorry for the delay in investigating. I finally have some time to devote to this issue. I tried to reproduce the problem to define the extent of this bug, and found the following:

- If the disconnected client is a WebSocket client, you can emit from a disconnect handler without seeing this issue.
- If the disconnected client is a long-polling client, and the client explicitly disconnects (by calling `disconnect()` on the javascript socket.io client), then this problem does not occur.
- If the disconnected client is a long-polling client, and the client just goes away (i.e. you close the browser or tab), then after about a minute a infinite recursion error like yours above occurs.

Does this match your situation?

**Comment by fake-name** [comment]
I'm explicitly disallowing long-polling clients, so I /should/ only have websocket clients. 

It may be worth noting that I'm pushing the websocket link pretty hard, the way the thing is used is that I effectively saturate the link between the server and the client: The client requests data, the server responds, and as soon as the client has received an update, it immediately requests data again, and this is done in two different callbacks for different data objects under different event names. 

Additionally, each request if for non-trivial data chunks, generally a few hundred kilobytes, which are also using the binary transport mode.

The work-around of using `include_self=False` seems to at least cover up the issue for the moment, so it's not urgent, but thanks for the ongoing interest.

**Comment by miguelgrinberg** [comment]
Fixed in python-socketio package, version 1.1. Please let me know if you continue to see problems after upgrade to that release.
